handler.go

```diff
@@ -150,41 +133,137 @@ func (h *Handler) makeAuthenticationHandler(fn func(http.ResponseWriter, *http.R
+	// without authentication -- the creation of the first admin user.
+	if h.AuthenticationEnabled && len(h.server.Users()) == 0 {
+		if len(query.Statements) == 1 {
+			c, ok := query.Statements[0].(*influxql.CreateUserStatement)
+			if !ok || c.Privilege == nil || (c.Privilege != nil && *c.Privilege != influxql.AllPrivileges) {
+				h.error(w, "initial admin user does not exist", http.StatusUnauthorized)
+				return
+			}
+		} else {
+			h.error(w, "initial admin user must be created", http.StatusUnauthorized)
 			return
 		}
-	*/
+	}
 
-	// Parse the time precision from the query params.
-	/*
-		precision, err := parseTimePrecision(urlQry.Get("time_precision"))
-		if err != nil {
-			h.error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-	*/
+	for _, s := range query.Statements {
+		switch c := s.(type) {
+		case *influxql.CreateDatabaseStatement:
+			if err = h.server.CreateDatabase(c.Name); err != nil {
+				h.error(w, "error creating database: "+err.Error(), http.StatusInternalServerError)
+				continue
+			}
+			w.WriteHeader(http.StatusCreated)
+		case *influxql.DropDatabaseStatement:
+			if err = h.server.DeleteDatabase(c.Name); err == ErrDatabaseNotFound {
+				h.error(w, "error deleting database: "+err.Error(), http.StatusInternalServerError)
+				continue
+			}
+			w.WriteHeader(http.StatusNoContent)
+		case *influxql.ListDatabasesStatement:
+			databases := h.server.Databases()
+			if databases != nil {
+				_ = json.NewEncoder(w).Encode(databases)
+			} else {
+				_ = json.NewEncoder(w).Encode([]string{})
+			}
 
-	// Execute query against the database.
-	/*
-		if err := db.ExecuteQuery(q); err != nil {
-			h.error(w, err.Error(), http.StatusInternalServerError)
-			return
+		case *influxql.CreateUserStatement:
+			isAdmin := false
+			if c.Privilege != nil {
+				isAdmin = *c.Privilege == influxql.AllPrivileges
+			}
+			if err := h.server.CreateUser(c.Name, c.Password, isAdmin); err != nil {
+				h.error(w, "error creating user: "+err.Error(), http.StatusInternalServerError)
+				continue
+			}
+			w.WriteHeader(http.StatusCreated)
+		case *influxql.DropUserStatement:
+			if err := h.server.DeleteUser(c.Name); err != nil {
+				h.error(w, "error deleting user: "+err.Error(), http.StatusInternalServerError)
+				continue
+			}
+			w.WriteHeader(http.StatusNoContent)
+
+		case *influxql.SelectStatement:
+			continue
+		case *influxql.DropSeriesStatement:
+			continue
+		case *influxql.ListSeriesStatement:
+			continue
+		case *influxql.ListMeasurementsStatement:
+			continue
+		case *influxql.ListTagKeysStatement:
+			continue
+		case *influxql.ListTagValuesStatement:
+			continue
+		case *influxql.ListFieldKeysStatement:
+			continue
+		case *influxql.ListFieldValuesStatement:
+			continue
+
+		case *influxql.GrantStatement:
+			continue
+		case *influxql.RevokeStatement:
+			continue
+
+		case *influxql.CreateRetentionPolicyStatement:
+			rp := NewRetentionPolicy(c.Name)
```

Comment:
The `switch` will have a lot of cases but if the body of each one is a one line function call then it's still manageable. 

Also, I didn't realize what file this one when I was reviewing but we'll need to move this `switch` into `Server` since the `Handler` shouldn't own the query execution.